feat: initial changes to show erc20 as native token in explorer

fkrause98 opened the pull request:
# What ❔
Update the explorer to show the ERC20 as the explorer's native token.
<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->
Closes #1.
## Why ❔
To reflect the current state of the chain.
<!-- Why are these changes done? What goal do they contribute to? What are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future iterators are in context about the evolution of repos. -->
## Setup and first check
### Start the zk server
Go to our fork repo of zksync-era and checkout to `kl-factory`, run: 
```
  zk init --base-token-name BAT
``` 
After it finishes, start the server:
``` 
zk server
``` 
### Explorer setup
1. Install dependencies: 
``` 
  npm i 
```
2. Build the project:
```
   npm run build
```
3. Set these env vars:
```
export DATABASE_HOST="127.0.0.1"
export BLOCKCHAIN_RPC_URL="http://127.0.0.1:3050"
export DATABASE_URL="postgres://postgres:postgres@127.0.0.1:5432/block-explorer"
```
4. Create the db:
```
   npm run db:create
```
5. Start the server:
```
   npm run dev
```
Go to http://localhost:3010/ , the block explorer should be up and running, 
but it probably won't be showing any transactions for now.

You should see something like this:
![image](https://github.com/lambdaclass/block-explorer/assets/56402156/43562c34-4fa2-4aa5-b639-1dc7db546f87)
## Setup and second check.
Do the steps above again, but run zk init without any other argument, the block
explorer should show ETH as the native token when doing a transaction.
## Checklist
- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

fkrause98 committed:
feat: initial changes to show erc20 as native token in explorer

fkrause98 committed:
feat: use nativechain instead of eth

fkrause98 committed:
feat: finishing touches to show ERC20 as native token

fkrause98 committed:
chore: rename .png

fkrause98 committed:
chore: rename png asset

fkrause98 committed:
feat: default to ethereum when rpc method is missing

fkrause98 committed:
fix: for error of missing method

juan518munoz reviewed (dismissed):
LGTM
<img width="1296" alt="image" src="https://github.com/lambdaclass/block-explorer/assets/62400508/504c76f0-5f48-413c-84f1-481fd2aafbaf">

fkrause98 committed:
fix: worker tests

fkrause98 committed:
fix: worker tests

fkrause98 committed:
fix: api tests and some e2e imports

fkrause98 committed:
fix: move token.ts file to work for imports

fkrause98 committed:
fix: common imports

fkrause98 committed:
feat: adapt to base-token-addr-endpoint

fkrause98 committed:
fix: worker tests

fkrause98 committed:
Merge branch 'main' into show-native-erc20token-info

fkrause98 committed:
fix: fix data-fetcher tests

fkrause98 committed:
chore: remove unused png

fkrause98 committed:
chore: replace "Native Token" with "Base Token"

fkrause98 committed:
chore: rename chainNative to baseToken

fkrause98 committed:
feat: move base token fetching to separate module

fkrause98 committed:
chore: remove old, unused code

fkrause98 committed:
fix: use base token instead of base token

fkrause98 committed:
fix: rename native token to base token

fkrause98 committed:
chore: remove old, unused code

fkrause98 committed:
fix: fix build, add comment

fkrause98 committed:
fix: restore postgres default password on config

fkrause98 committed:
feat: use base token from config in more places

fkrause98 committed:
chore: update .env.example

fkrause98 committed:
feat: move base-token fetching to a script

fkrause98 committed:
fix: data-fetcher unit tests

fkrause98 committed:
fix: api unit tests

fkrause98 committed:
fix: worker tests

fkrause98 committed:
fix: more tests

fkrause98 committed:
chore: properly import config

fkrause98 committed:
feat: rename to BaseToken

fkrause98 committed:
feat: remove baseTokenData function

fkrause98 committed:
feat: add base token address to configs

fkrause98 committed:
feat: use base token from config instead of constant

fkrause98 committed:
chore: remove unused scriped script

fkrause98 committed:
Merge branch 'main' into show-native-erc20token-info

Commit:
fix: update baseToken to baseTokenData in E2E tests

Commit:
fix(e2e tests): fix end-to-end test to use base token, modify eth address

Commit:
fix(e2e tests): fix eth address in address e2e tests

Commit:
fix(e2e tests): change missing apparences of l1 address

Commit:
fix(e2e tests): non existing var name

vasyl-ivanchuk committed:
fix: increase default timeout settings (#228)

# What ❔

Increase default timeout settings.

## Why ❔

To make sure explorer can handle large transactions.

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [X] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [X] Tests for the changes have been added / updated.

SantiagoPittella committed:
Merge branch 'main' into show-native-erc20token-info

Commit:
fix(e2e tests): valid address format

Commit:
fix(e2e tests): add missing fields in select

SantiagoPittella committed:
fix(e2e tests): remove unused modules

SantiagoPittella committed:
fix(e2e tests): typecheck

SantiagoPittella committed:
fix(e2e tests): increase amount of selects in test

SantiagoPittella committed:
fix(e2e tests): add missing values in tests

SantiagoPittella committed:
test(coverage): api config with base token

SantiagoPittella committed:
test: uncomment tests

SantiagoPittella committed:
fix: remove non existing script

SantiagoPittella committed:
fix: address PR comments

SantiagoPittella committed:
fix: lint errors

SantiagoPittella committed:
fix: rpc method name

SantiagoPittella committed:
fix: rollback address capitalization

vasyl-ivanchuk committed:
fix: rename root hash to block hash (#249)

# What ❔

Rename root hash on the block screen to block hash.

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [X] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [X] Tests for the changes have been added / updated.

SantiagoPittella committed:
fix: use checksum address

SantiagoPittella committed:
Merge branch 'main' into show-native-erc20token-info

SantiagoPittella committed:
Merge branch 'show-native-erc20token-info' of github.com:lambdaclass/block-explorer into show-native-erc20token-info

SantiagoPittella committed:
fix: use both version of the rpc response for getBridgeContract method

fkrause98 committed:
Revert "fix(e2e tests): add missing fields in select"

This reverts commit d3916ed83c87133e0b4fa8bde7e9e871bf66001d.

SantiagoPittella committed:
Revert "fix(e2e tests): add missing fields in select"

This reverts commit d3916ed83c87133e0b4fa8bde7e9e871bf66001d.

fkrause98 committed:
fix(e2e): account tests

SantiagoPittella committed:
Merge branch 'show-native-erc20token-info' of github.com:lambdaclass/block-explorer into show-native-erc20token-info

SantiagoPittella committed:
fix: use correct amount of selects

SantiagoPittella committed:
test: add tests for old response

vasyl-ivanchuk committed:
chore: fix docker compose (#256)

# What ❔

Fix docker compose by applying the same updates that were done in
[local-setup](https://github.com/matter-labs/local-setup.)

## Why ❔

Old docker compose doesn't work with the latest `local-node` docker
image.

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [X] PR title corresponds to the body of PR (we generate changelog
entries from PRs).

vasyl-ivanchuk committed:
fix: explorer doesn't require defined bridge address (#257)

# What ❔

Fix explorer so that a defined bridge address is not required.

## Why ❔

In-memory node doesn't have L1 network and bridge addresses defined.
This fix make explorer work with the in-memory node.

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [X] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [X] Tests for the changes have been added / updated.

SantiagoPittella committed:
Merge branch 'main' into show-native-erc20token-info
